### PR TITLE
Fix: replace as const objects with enum

### LINKED ISSUE #1768: Use enums instead of `as const` objects
This issue is to replace all occurrences of `as const` objects with string enums.

Example:
```ts
export const TOKENS_MODAL_TYPES = {
  activate: 'activate',
  deactivate: 'deactivate',
} as const;
```

should become:
```ts
export enum TokensModalType {
  Activate = 'activate',
  Deactivate = 'deactivate'
}
```

### PR DESCRIPTION
**Pull request by warrencolliver**
## Description

This PR replaces all as const objects to enum objects including the imports in other components. Due to the enums being used across the app there have been changes in many files.

There is no changes to the UI in this PR. 

**Testing**

1. Go through code changes in the files
2. Perform actions in the UI which use these objects. e.g manage permissions action

## Diffs

**Changes** 🏗

as const objects which have been changed:
DISTRIBUTION_METHOD
AUTHORITY
FIELD_STATE
TOKENS_MODAL_TYPES
STATUS_TYPES
ACTION
USER_ROLE
VOTE_CHART_BAR_DIRECTION

Resolves #1768

### CODE REVIEW

File: src/constants/actions.ts [hunk 1]
```diff
@@ -1,26 +1,24 @@
...
+  EDIT_EXISTING_TEAM = 'edit-existing-team',
+  MANAGE_REPUTATION = 'manage-reputation',
+  MANAGE_PERMISSIONS = 'manage-permissions',
+  EDIT_COLONY_DETAILS = 'edit-colony-details',
+  UPGRADE_COLONY_VERSION = 'upgrade-colony-version',
+  ENTER_RECOVERY_MODE = 'enter-recovery-mode',
+  CREATE_NEW_INTEGRATION = 'create-new-integration',
+  MANAGE_COLONY_OBJECTIVES = 'manage-colony-objectives',
+  USER_PERMISSIONS = 'user-permissions',
+  SIMPLE_DISCUSSION = 'simple-discussion',
```
Comment (rdig) [inline src/constants/actions.ts:23]: I'd convert these to camel case as well. 

Ex:
```
SimplePayment = 'simple-payment',
```

File: src/stories/shared/StatusText.stories.tsx [hunk 2]
```diff
@@ -23,7 +24,7 @@ const statusTextMeta: Meta<typeof StatusText> = {
     },
   },
   args: {
-    status: 'success',
+    status: StatusTypes.SUCCESS,
```
Comment (rdig) [inline src/stories/shared/StatusText.stories.tsx:27]: As, with actions, these should be CamelCase as well

```suggestion
    status: StatusTypes.Success,
```

**Review by rdig — APPROVED** [review]
Builds fine, no errors in the console, and no obvious regressions in the UI.

Left some comments I want you to tackle first, but otherwise nice work!

Screenshot from 2024-02-16 18-20-02
Screenshot from 2024-02-16 18-20-25

**Review by ArmandoGraterol — APPROVED** [review]
LGTM 👌 Nice work Warren!

**Pull request merged**